src/App.jsx

```diff
@@ -45,14 +59,25 @@ class App extends Component {
   }
 
   runPostLogin() {
-    this.setState({ isLoggedIn: true });
-    GitHubService.getFile(this.props.config.repo).then((source) => {
-      DocumentService.setSource(source);
-      this.setState({ content: DocumentService.getJSON() });
+    this.setState({
+      isLoggedIn: true,
+      isLoading: true,
+    });
+    GitHubService.loadSchema(this.props.config).then((schema) => {
+      store.dispatch({ type: 'SET_SCHEMA', value: schema });
+      GitHubService.getFile(this.props.config.repo).then((source) => {
+        DocumentService.setSource(source);
```

Comment:
Actually you should also be mapping dispatch, to props methods. See here: http://redux.js.org/docs/basics/UsageWithReact.html

It looks like you are moving in this direction, but good to clarify, basically you you should have high level containers, that mapstate to props and mapDispatch to methods. You should then just pass those props and methods to child (presentational components if needed). 

Good examples are here: http://redux.js.org/docs/basics/UsageWithReact.html